web/src/unread.js

```diff
@@ -709,10 +709,13 @@ export function update_message_for_mention(message, content_edited = false) {
         return false;
     }
 
+    // A message is said to have an unmuted mention if message contains a mention and
+    // if the message is a direct message or if the message is in an unmuted topic in an unmuted stream.
     const is_unmuted_mention =
-        message.type === "stream" &&
         message.mentioned &&
-        !user_topics.is_topic_muted(message.stream_id, message.topic);
+        (message.type === "private" ||
+            (!stream_data.is_muted(message.stream_id) &&
+                !user_topics.is_topic_muted(message.stream_id, message.topic)));
```

Comment:
@prakhar1144 can you help review this logic? I'm a bit worried that it doesn't handle followed/unmuted topics properly.

Reply:
A message is unmuted mention if it belongs to either:
* not muted topic in unmuted stream
* unmuted or followed topic in muted stream

and have `mentioned` flag.

**The logic above doesn't covers the second case:**
* message in "unmuted or followed topic in muted stream"

Reply:
@prakhar1144, thanks for pointing out this missed case of if the message is in "unmuted or followed topic in a muted stream".

So, any direct mention in the muted stream will still be counted as a mention due to this `if` condition:
`if (is_unmuted_mention || message.mentioned_me_directly) {` as `mentioned_me_directly` flag will be true for that message.

But, for wildcard mentions(`@all` and `@everyone`), any wildcard mention in a muted stream will not be counted as mentions even if the `mentioned` flag is `true` as per current logic.

Just wanted to confirm that do we count wildcard mentions in "unmuted or followed topic in a muted stream" as mentions?

If so, I will update the `is_unmuted_mention` code to be this so that wildcard mentions in "unmuted or followed topic in a muted stream" get counted as mentions:

```
// A message is said to have an unmuted mention if message contains a mention and
    // if the message is a direct message or
    // if the message is in a not muted topic in an unmuted stream or
    // if the message is in a followed or an unmuted topic in a muted stream.
    const is_unmuted_mention =
        message.mentioned &&
        (message.type === "private" ||
            (!stream_data.is_muted(message.stream_id) &&
                !user_topics.is_topic_muted(message.stream_id, message.topic)) ||
            (stream_data.is_muted(message.stream_id) &&
                user_topics.is_topic_unmuted_or_followed(message.stream_id, message.topic)));
```

Reply:
> Just wanted to confirm that do we count wildcard mentions in "unmuted or followed topic in a muted stream" as mentions?

Yes